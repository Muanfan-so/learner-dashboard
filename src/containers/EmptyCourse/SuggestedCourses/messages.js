import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  header: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.header',
    defaultMessage: 'Some courses you may be interested in',
    description: 'Header for suggesting courses',
  },
  viewCourseButton: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.viewCourseButton',
    defaultMessage: 'View Course',
    description: 'Actions button to visit the suggested course',
  },
  courseImageAlt: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.courseImageAlt',
    defaultMessage: 'Course image banner',
    description: 'alt for course image',
  },
  institueLogoAlt: {
    id: 'leanerDashboard.emptyCourse.suggestedCourses.institueLogoAlt',
    defaultMessage: 'Course institute logo',
    description: 'alt for institute logo',
  },
});

export default messages;
