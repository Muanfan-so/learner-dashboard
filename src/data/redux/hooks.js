import { useSelector } from 'react-redux';

import { actions as appActions } from './app/reducer';
import appSelectors from './app/selectors';
import requestSelectors from './requests/selectors';

const { courseCard } = appSelectors;

export const usePageNumber = () => useSelector(appSelectors.pageNumber);
export const useEmailConfirmationData = () => useSelector(appSelectors.emailConfirmation);
export const useEnterpriseDashboardData = () => useSelector(appSelectors.enterpriseDashboard);
export const usePlatformSettingsData = () => useSelector(appSelectors.platformSettings);
// suggested courses is max at 3 at the moment.
export const useSuggestedCoursesData = () => useSelector(appSelectors.suggestedCourses).slice(0, 3);
export const useRecommendedCoursesData = () => useSelector(appSelectors.recommendedCourses);
export const useSelectSessionModalData = () => useSelector(appSelectors.selectSessionModal);

export const useHasCourses = () => useSelector(appSelectors.hasCourses);
export const useHasAvailableDashboards = () => useSelector(appSelectors.hasAvailableDashboards);
export const useCurrentCourseList = (opts) => useSelector(
  state => appSelectors.currentList(state, opts),
);
export const useShowSelectSessionModal = () => useSelector(appSelectors.showSelectSessionModal);

// eslint-disable-next-line
export const useCourseCardData = (selector) => (cardId) => useSelector(
  (state) => selector(state, cardId),
);

export const useCardCertificateData = useCourseCardData(courseCard.certificate);
export const useCardCourseData = useCourseCardData(courseCard.course);
export const useCardCourseRunData = useCourseCardData(courseCard.courseRun);
export const useCardEnrollmentData = useCourseCardData(courseCard.enrollment);
export const useCardEntitlementData = useCourseCardData(courseCard.entitlement);
export const useCardGradeData = useCourseCardData(courseCard.gradeData);
export const useCardProviderData = useCourseCardData(courseCard.courseProvider);
export const useCardRelatedProgramsData = useCourseCardData(courseCard.relatedPrograms);

export const useUpdateSelectSessionModalCallback = (dispatch, cardId) => () => dispatch(
  appActions.updateSelectSessionModal(cardId),
);

export const useMasqueradeData = () => useSelector(requestSelectors.masquerade);

export const useIsPendingRequest = (requestName) => useSelector(requestSelectors.isPending(requestName));
