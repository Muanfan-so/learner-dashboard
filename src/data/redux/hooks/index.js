import * as app from './app';
import * as requests from './requests';

export * from './app';
export { requests };

export default { app, requests };
