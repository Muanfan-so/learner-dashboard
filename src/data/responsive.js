import { useWindowSize, breakpoints } from '@edx/paragon';
import { StrictDict } from 'utils';

const createResponsive = (fn) => () => {
  const { width } = useWindowSize();
  return fn(width);
};

export const isDesktopSize = createResponsive(
  (width) => width >= breakpoints.large.minWidth,
);

// Beware of using isPadSize() for styling. When the screen resizes pass the
// breakpoint, the screen will flash between the two styles. This is why it is
// recommened isDesktopSize() or isMobileSize() be used instead.
export const isIpadSize = createResponsive(
  (width) => width >= breakpoints.medium.minWidth && width < breakpoints.large.minWidth,
);

export const isMobileSize = createResponsive(
  (width) => width < breakpoints.medium.minWidth,
);

export default StrictDict({
  isDesktopSize,
  isIpadSize,
  isMobileSize,
});
