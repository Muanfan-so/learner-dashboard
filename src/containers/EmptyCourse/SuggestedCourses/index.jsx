import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Card, Button, Container } from '@edx/paragon';
import { hooks as appHooks } from 'data/redux';

import messages from './messages';

export const SuggestedCourses = () => {
  const suggestedCourses = appHooks.useSuggestedCoursesData();
  const { formatMessage } = useIntl();
  if (!suggestedCourses.length) { return null; }
  return (
    <Container size="md">
      <h3 className="text-center">{formatMessage(messages.header)}</h3>
      <div className="d-flex">
        {suggestedCourses.map((course) => (
          <Card key={course.title} className="m-3">
            <Card.ImageCap
              src={course.bannerUrl}
              srcAlt={formatMessage(messages.courseImageAlt)}
              logoSrc={course.logoUrl}
              logoAlt={formatMessage(messages.institueLogoAlt)}
            />
            <Card.Header title={course.title} />
            <Card.Footer>
              <Button as="a">{formatMessage(messages.viewCourseButton)}</Button>
            </Card.Footer>
          </Card>
        ))}
      </div>
    </Container>
  );
};

export default SuggestedCourses;
