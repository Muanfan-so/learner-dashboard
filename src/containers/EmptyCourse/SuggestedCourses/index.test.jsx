import React from 'react';
import { shallow } from 'enzyme';

import { hooks as appHooks } from 'data/redux';
import SuggestedCourses from '.';

jest.mock('data/redux', () => ({
  hooks: {
    useSuggestedCoursesData: jest.fn(),
  },
}));

const suggestedCourses = [
  {
    bannerUrl: 'https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg',
    logoUrl: 'https://prod-discovery.edx-cdn.org/organization/certificate_logos/b9dc96da-b3fc-45a6-b6b7-b8e12eb79335-ac60112330e3.png',
    title: 'Suggested course 1',
    courseUrl: 'www.edx/suggested-course',
  },
  {
    bannerUrl: 'https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg',
    logoUrl: 'https://prod-discovery.edx-cdn.org/organization/certificate_logos/b9dc96da-b3fc-45a6-b6b7-b8e12eb79335-ac60112330e3.png',
    title: 'Suggested course 2',
    courseUrl: 'www.edx/suggested-course',
  },
  {
    bannerUrl: 'https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg',
    logoUrl: 'https://prod-discovery.edx-cdn.org/organization/certificate_logos/b9dc96da-b3fc-45a6-b6b7-b8e12eb79335-ac60112330e3.png',
    title: 'Suggested course 3',
    courseUrl: 'www.edx/suggested-course',
  },
];

describe('SuggestedCourses snapshot', () => {
  test('no suggested courses', () => {
    appHooks.useSuggestedCoursesData.mockReturnValueOnce([]);
    expect(shallow(<SuggestedCourses />)).toMatchSnapshot();
  });

  test('has 3 suggested courses', () => {
    appHooks.useSuggestedCoursesData.mockReturnValueOnce(suggestedCourses);
    expect(shallow(<SuggestedCourses />)).toMatchSnapshot();
  });
});
