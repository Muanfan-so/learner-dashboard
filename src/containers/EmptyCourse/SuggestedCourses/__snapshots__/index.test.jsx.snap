// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedCourses snapshot has 3 suggested courses 1`] = `
<Container
  size="md"
>
  <h3
    className="text-center"
  >
    Some courses you may be interested in
  </h3>
  <div
    className="d-flex"
  >
    <Card
      className="m-3"
      key="Suggested course 1"
    >
      <Card.ImageCap
        src="https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg"
        srcAlt="Course image banner"
      />
      <Card.Header
        className="mb-2"
        title="Suggested course 1"
      />
      <Card.Body
        className="h-100"
      />
      <Card.Footer>
        <ActionRow>
          <Button
            as="a"
          >
            View Course
          </Button>
        </ActionRow>
      </Card.Footer>
    </Card>
    <Card
      className="m-3"
      key="Suggested course 2"
    >
      <Card.ImageCap
        src="https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg"
        srcAlt="Course image banner"
      />
      <Card.Header
        className="mb-2"
        title="Suggested course 2"
      />
      <Card.Body
        className="h-100"
      />
      <Card.Footer>
        <ActionRow>
          <Button
            as="a"
          >
            View Course
          </Button>
        </ActionRow>
      </Card.Footer>
    </Card>
    <Card
      className="m-3"
      key="Suggested course 3"
    >
      <Card.ImageCap
        src="https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg"
        srcAlt="Course image banner"
      />
      <Card.Header
        className="mb-2"
        title="Suggested course 3"
      />
      <Card.Body
        className="h-100"
      />
      <Card.Footer>
        <ActionRow>
          <Button
            as="a"
          >
            View Course
          </Button>
        </ActionRow>
      </Card.Footer>
    </Card>
  </div>
</Container>
`;

exports[`SuggestedCourses snapshot no suggested courses 1`] = `""`;
